nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/actions/RemoveProxyAction.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * (C) Copyright 2018 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Funsho David
+ */
+
+package org.nuxeo.ecm.core.bulk.actions;
+
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.DocumentModelList;
+import org.nuxeo.ecm.core.api.DocumentRef;
+import org.nuxeo.ecm.core.api.IdRef;
+import org.nuxeo.lib.stream.computation.Computation;
+
+/**
+ * @since 10.3
+ */
+public class RemoveProxyAction extends AbstractBulkAction {
+
+    public static final String ACTION_NAME = "removeProxy";
+
+    public RemoveProxyAction() {
+        super(ACTION_NAME);
+    }
+
+    @Override
+    protected Computation createComputation(int batchSize, int batchThresholdMs) {
+        return new RemoveProxyComputation(getActionName(), batchSize, batchThresholdMs);
+    }
+
+    public static class RemoveProxyComputation extends AbstractBulkComputation {
+
+        public RemoveProxyComputation(String name, int batchSize, int batchThresholdMs) {
+            super(name, 1, 1, batchSize, batchThresholdMs);
```

Comment:
Just an observation. This computation takes 1 input stream and the `AbstractBulkComputation` writes to 1 output stream, which is the counter.  Is that right?

Reply:
yes, the 1 ouput is count, but that's a good question, should we compute this value in accordance of doesCount ?

Reply:
if this is the convention used in a bulk computation, it should be stated somewhere